Sources/AppKit/NSTableView.swift

```diff
@@ -36,20 +37,164 @@ public extension ReactiveExtensions where Base: NSTableView {
+}
 
-  @discardableResult
-  public func bind(to tableView: NSTableView, animated: Bool = true, createCell: @escaping (DataSource, Int, NSTableView) -> NSView?) -> Disposable {
+public protocol TableViewBond: TableViewBondOptionable {
+	
+	associatedtype DataSource: DataSourceProtocol
+	
+	func apply(event: DataSourceEvent<DataSource>, to tableView: NSTableView)
+	func heightForRow(at index: Int, tableView: NSTableView, dataSource: DataSource) -> CGFloat?
+	func cellForRow(at index: Int, tableView: NSTableView, dataSource: DataSource) -> NSView?
+}
+
+extension TableViewBond {
+	
+	public var insertAnimation: NSTableViewAnimationOptions? {
+		return [.effectFade, .slideUp]
+	}
+	
+	public var deleteAnimation: NSTableViewAnimationOptions? {
+		return [.effectFade, .slideUp]
+	}
+	
+	public func heightForRow(at index: Int, tableView: NSTableView, dataSource: DataSource) -> CGFloat? {
+		return nil
+	}
+	
+	public func apply(event: DataSourceEvent<DataSource>, to tableView: NSTableView) {
+		if insertAnimation == nil && deleteAnimation == nil {
+			tableView.reloadData()
+			return
+		}
+		
+		switch event.kind {
+		case .reload:
+			tableView.reloadData()
+		case .insertItems(let indexPaths):
+			if !tableView.updating && indexPaths.count > 1 {
+				tableView.beginUpdates()
+				defer { tableView.endUpdates() }
+			}
+			indexPaths.forEach { indexPath in
+				tableView.insertRows(at: IndexSet(integer: indexPath.item), withAnimation: insertAnimation ?? [])
+			}
+		case .deleteItems(let indexPaths):
+			if !tableView.updating && indexPaths.count > 1 {
+				tableView.beginUpdates()
+				defer { tableView.endUpdates() }
+			}
+			indexPaths.forEach { indexPath in
+				tableView.removeRows(at: IndexSet(integer: indexPath.item), withAnimation: deleteAnimation ?? [])
+			}
+		case .reloadItems(let indexPaths):
+			if !tableView.updating && indexPaths.count > 1 {
+				tableView.beginUpdates()
+				defer { tableView.endUpdates() }
+			}
+			indexPaths.forEach { indexPath in
+				tableView.removeRows(at: IndexSet(integer: indexPath.item), withAnimation: deleteAnimation ?? [])
+				tableView.insertRows(at: IndexSet(integer: indexPath.item), withAnimation: insertAnimation ?? [])
+			}
+		case .moveItem(let indexPath, let newIndexPath):
+			tableView.moveRow(at: indexPath.item, to: newIndexPath.item)
+		case .insertSections:
+			fatalError("NSTableView binding does not support sections.")
+		case .deleteSections:
+			fatalError("NSTableView binding does not support sections.")
+		case .reloadSections:
+			fatalError("NSTableView binding does not support sections.")
+		case .moveSection:
+			fatalError("NSTableView binding does not support sections.")
+		case .beginUpdates:
+			tableView.beginUpdates()
+			tableView.updating = true
+		case .endUpdates:
+			tableView.updating = false
+			tableView.endUpdates()
+		}
+	}
+}
+
+// MARK: - Builtin bonds
+
+private struct DefaultTableViewBond<DataSource: DataSourceProtocol>: TableViewBond {
```

Comment:
Since we are moving `updating` here, it should be a class. I would also make it `open class` so that it can be subclassed by someone who wants to extend the default behaviour.